### ISSUE TITLE
[27.2] The text area is getting smaller and smaller than 27.1.3

### ISSUE THREAD

**Issue by iesv** [obsproject/obs-studio#5910]
### Operating System Info

Windows 10

### OBS Studio Version

27.2.0 Beta 4

### OBS Studio Log URL

https://obsproject.com/logs/QMXmDkcbbTOBe2uO

### Expected Behavior

Remember the resized window.

### Current Behavior

The preview window is too big and wastes a lot of space, but the commonly used editing area is too small, so that it's not convenient to edit or adjust text. And the default text area is too far to the bottom and not convenient for writing. I always have to resize it manually every time. The user experience is not good. Is it possible to remember the resized window?
![11](https://user-images.githubusercontent.com/16969250/152282376-be6416fd-5e45-4172-98c3-d1128c86db7d.png)

### Steps to Reproduce

1. Add a text source
2. Click the Properties

**Comment by Fenrirthviti** [comment]
There is a separator between the text preview and the options, simply drag it to make the preview larger or smaller as you want.

**Comment by WizardCM** [comment]
For reference, I did some digging for this 2 days ago.

1. This code says that the view should be a 3:1 stretch <https://github.com/obsproject/obs-studio/blob/95020dd62923e42847db87d3ffb164d05874cdef/UI/window-basic-properties.cpp#L111-L112>
2. However, no matter how tall you open the Properties window, the properties section always has the exact same height
3. I _think_ this is due to the Preview requesting a full stretch without the parent view knowing how tall it is properly.
If I completely remove the stretch factor & `preview->sizePolicy` it behaves _better_, but still not 3:1 or anything useful.

**Comment by Fenrirthviti** [comment]
Ah, sorry, I see what you mean now, the size is not being saved. Can you confirm if this is new to 27.2 or if you experienced this in previous versions?

**Comment by iesvme** [comment]
> For reference, I did some digging for this 2 days ago.
> 
> 1. This code says that the view should be a 3:1 stretch https://github.com/obsproject/obs-studio/blob/95020dd62923e42847db87d3ffb164d05874cdef/UI/window-basic-properties.cpp#L111-L112
> 2. However, no matter how tall you open the Properties window, the properties section always has the exact same height
> 3. I _think_ this is due to the Preview requesting a full stretch without the parent view knowing how tall it is properly.
> 
> If I completely remove the stretch factor & `preview->sizePolicy` it behaves _better_, but still not 3:1 or anything useful.
> 
> I don't believe this is new to 27.2 though.

Thanks for your reply.
It would also be nice if the default ratio was adjusted to a 1:1 size.

**Comment by WizardCM** [comment]
In my opinion the ratio should be highly dependent on the size of the preview section. If the source is wider than it is tall, then the properties should take up more space (as is the case with the text source). For sources that are tall but also have a lot of properties, I think 1:1 or 2:1 would be best.

**Comment by iesvme** [comment]
And I thought we were usually working on the editing area, and the preview window was sufficient with a small window.

**Comment by ebot102** [comment]
Hey there! I'm new to the project and would like to try and tackle this issue. Lines 111-112 seemed to have moved from [obs-studio/UI/window-basic-properties.cpp](https://github.com/obsproject/obs-studio/blob/95020dd62923e42847db87d3ffb164d05874cdef/UI/window-basic-properties.cpp#L111-L112) in the master branch. Any advice on where to start looking?

**Comment by WizardCM** [comment]
@ebot102 That is due to the merging of #6088, I expect the code to look at is not in the Properties window at all, but rather something deep in https://github.com/obsproject/obs-studio/blob/master/UI/qt-display.cpp where it adapts its size to the content regardless of the container it's in.

**Comment by ebot102** [comment]
So I see where it is resizing the content in [https://github.com/obsproject/obs-studio/blob/master/UI/qt-display.cpp](https://github.com/obsproject/obs-studio/blob/master/UI/qt-display.cpp) (lines 213-216)
```
if (isVisible() && display) {
		QSize size = GetPixelSize(this);
		obs_display_resize(display, size.width(), size.height());
	}
```
However, the only place I have been able to find that the scaling ratios are set is in [https://github.com/obsproject/obs-studio/blob/master/UI/forms/OBSBasicProperties.ui](https://github.com/obsproject/obs-studio/blob/master/UI/forms/OBSBasicProperties.ui) under the "propertiesFrame" tag. Is this the correct place to edit the ratio for text boxes, or is this defining the scaling ratios for the parent container?

**Comment by gxalpha** [comment]
Fixed by #6671
